### ISSUE TITLE
"Is not a valid floating point value" error

### ISSUE THREAD

**Issue by Brawdogger** [weathondev/SingstarCreatorTutorial#8]
![Floatingpointerror](https://github.com/user-attachments/assets/b1d6d7af-a53d-4cc1-a123-b16275cccb96)
When verifying the txt files of some songs, I receive this error which when translated is referencing a floating point value not being valid. I suspect this issue may be caused by a note that whilst being valid in ultrastar, is not allowed within singstar and causes issues. Could a potential fix to be either creating a script to automatically move these notes to the nearest valid point or do them manually within ultrastar's editor?

**Comment by weathondev** [comment]
Hey! As you have a German os I suspect that those notes were created on a German os as well (the comma in the float value suggests it too).
As in many German speaking countries at least we use a , instead of a .
Try and see if you can just replace all commas with dots (depends what else is in the txt)

**Comment by Brawdogger** [comment]
Nah I'm not actually on a german OS, singstar creator is just programmed to give these error messages in german regardless of what country you're in.
Am currently talking with MinorProblems and have been able to figure out what might be causing this issue. It appears singstar doesn't accept BPM's that aren't whole numbers and will give errors if it has a decimal point. Removing any number past the comma no longer produces these errors. Could possibly cause some issues with the timing of songs but will have to test it to figure out if that's the case

**Comment by weathondev** [comment]
Ah makes sense. I think i encountered this as well. We'll need to add this to the repo so others don't need to troubleshoot as well.

**Comment by weathondev** [comment]
Btw. im pretty sure that Singstar is fine with floating values as soon as they are delimited by a dot . and not a comma ,

**Comment by Brawdogger** [comment]
Ayo has been quite a while since I ran into these issues but thought I should give you an update on how I resolved these issues. I decided to make a custom tool that can be used alongside Singstar Creator to scan these text files and find any errors that SC misses. Can be quite handy in troubleshooting any issues that caused SC to stop working. Give it a look and feel free to add it to your guide to help any future users: https://github.com/Brawdogger/Singstar-Creator-Text-File-Checker
